express-post-json/index.js

```diff
@@ -0,0 +1,28 @@
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+
+const grades = [];
+
+app.get('/api/grades', (req, res) => {
+  const gradesArr = [];
+  for (const id in grades) {
+    gradesArr.push(grades[id]);
+  }
+  res.json(gradesArr);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  grades.push(req.body);
+  grades[grades.length - 1].id = nextId;
+  nextId++;
+  res.sendStatus(201);
```

Comment:
With `grades` as an object you have to create a new property in the object using the `nextId` as the key. For example:
```JAVASCRIPT
app.post('/api/grades', (req, res) => {
  const newGrade = req.body;
  const id = nextId;
  newGrade.id = id;
  grades[id] = newGrade;
  nextId++;
  res.status(201).json(newGrade);
});
```

Reply:
Thank you! That helps a lot. It makes sense that it should be like the previous grades object.